lib/core_extensions.rb

```diff
@@ -39,8 +39,14 @@ def initialize(*attribute)
 
     def before_destroy(record)
       klasses.each do |klass|
-        record.send(klass.to_sym).each do |what|
-          what = what.to_label unless what.is_a? String
+        record.association(klass.to_sym).association_scope.each do |what|
```

Comment:
Ah okay, would you mind checking the Rails 4 behaviour so we can work out if this needs to change now, or only in the Rails 4 branch?  (Or if some other fix is needed there.)  As you'll gather, I don't wish to commit something that'll cause problems in the medium term.